Add 'gdal vector update' (#13412)
OSGeo/gdal: doc/source/programs/gdal_vector_update.rst

.. _gdal_vector_update:

================================================================================

> ``gdal vector update``

Author: kind of, except when you join you generally add extra information, whereas here in general the schemas of input and output layers will generally be the same

.. _gdal_vector_update:

================================================================================
``gdal vector update``
================================================================================

.. versionadded:: 3.13

.. only:: html

    Update an existing vector dataset with an input vector dataset

.. Index:: gdal vector update

Synopsis
--------

.. program-output:: gdal vector update --help-doc

Description
-----------

:program:`gdal vector update` can be used to update the features of an existing
target vector dataset with the features of the source vector. It identifies
matching features between input and output based on their feature ID by default,
or based on the value of one or several columns forming a key, when using :option:`--key`.

> With the default ``merge`` mode, features found in both layers will be updated,

Reviewer: ```suggestion
When using the default ``merge`` mode, features found in both layers will be updated,
```

and features only existing in the source layer will be created in the target layer.

> The schema of the source and target layers must not necessarily strictly match

Reviewer: ```suggestion
The schemas of the source and target layers do not need to be identical
```

Author: fixed
